python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -616,15 +955,18 @@ def validate(self):
             color1_errors = True
         elif ((self.color1_flux1_err and not self.color1_flux2_err)
               or (not self.color1_flux1_err and self.color1_flux2_err)):
-            raise ValueError("Must set both color1_flux1_err and color1_flux2_err if either is set.")
+            msg = "Must set both color1_flux1_err and color1_flux2_err if either is set."
+            raise FieldValidationError(self.__class__.color1_flux1_err, self, msg)
```

Comment:
I hijacked a few other commits into this ticket since these are probably not worth making a ticket on its own. Especially since @timj is looking at this PR :-)